lib/puppet/parser/scope.rb

```diff
@@ -870,7 +873,24 @@ def []=(varname, value, _ = nil)
 
   # Used mainly for logging
   def to_s
-    "Scope(#{@resource})"
+    # As this is used for logging, this should really not be done in this class at all...
+    return "Scope(#{@resource})" unless @resource.nil?
+
+    # For logging of function-scope - it is now showing the file and line.
+    detail = Puppet::Pops::PuppetStack.stacktrace[0]
+    return "Scope()" unless detail.is_a?(Array)
+
+    # shorten the path if possible
+    path = detail[0]
+    env_path = nil
+    env_path = environment.configuration.path_to_env unless (environment.nil? || environment.configuration.nil?)
+    if env_path && path && path.start_with?(env_path)
+      path = "<env>" + path[env_path.length..-1]
+    end
+    # TODO: also shorten module paths to <module>/name/...
```

Comment:
Is this still TODO?

Reply:
@hlindberg, this is from PUP-7945 "Make scope handle a nil path to env in its to_s method". Is there additional work that needs to be done before this is merged into master?

Reply:
It is not required. It is an opportunity to reduce logged output and increase readability when calls are made to logging functions (notice, debug, etc) from within plans in modules. It was marked as TODO as it is a bit of work to do that. Think we should log a ticket for this improvement.

Reply:
I logged ticket: https://tickets.puppetlabs.com/browse/PUP-8235 for this